pkgs/by-name/co/cosmic-comp/package.nix

```diff
@@ -57,10 +73,16 @@ rustPlatform.buildRustPackage {
   # be the common case, so just make them available, don't link them.
   postInstall = ''
     wrapProgram $out/bin/cosmic-comp \
-      --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [
-        xorg.libX11 xorg.libXcursor xorg.libXi xorg.libXrandr
-      ]} \
-      --prefix PATH : ${lib.makeBinPath [ xwayland ]}
+      --prefix LD_LIBRARY_PATH : ${
+        lib.makeLibraryPath [
+          xorg.libX11
+          xorg.libXcursor
+          xorg.libXi
+          xorg.libXrandr
+        ]
+      } \
```

Comment:
We don't know what line comes after this if we don't include the useXwayland stuff, so some unexpected line might end up getting added on to this one.

We could unconditionally insert a blank line after (or just in the non-Xwayland case), but it might be better to do what lots of other packages do and build up a shell array for wrapProgram args.

Reply:
Do you have a reference implementation of this? I'm not yet aware of this method.

Reply:
It'd look something like this:

```nix
''
wrapProgramArgs=(--prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [
  xorg.libX11 xorg.libXcursor xorg.libXi xorg.libXrandr
]})
'' + lib.optionalString useXWayland ''
wrapProgramArgs+=(--prefix PATH : ${lib.makeBinPath [ xwayland ]})
'' + ''
wrapProgram $out/bin/cosmic-comp "${wrapProgramArgs[@]}"
''
```